docs/application/native/guides/connectivity/softap.md

```diff
@@ -0,0 +1,341 @@
+   `is_requested` indicates which application has requested the change.
+
+   ```
+   static void __enabled_cb(softap_error_e error, bool is_requested, void *data)
+   {
+     if (error != SOFTAP_ERROR_NONE) {
+       if (!is_requested)
+         return;
+       printf("Failed to enable SoftAP[0x%X]", error);
+       return;
+   }   
+
+   if (is_requested)
+     printf("SoftAP is enabled successfully");
+   else
+     printf("SoftAP is enabled by other app");
+   }
+   ```
+
+To register the callback:	
+   ```
+   ret = softap_set_enabled_cb(softap, enabled_cb, NULL);                                 
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Fail to regester the callback.");
+   ```
+
+2. Define and register `disabled_cb()` callback. It is called when the SoftAP is disabled.
+   `code` indicates the cause of disabling.
+   ```
+   static void __disabled_cb(softap_error_e error, softap_disabled_cause_e code, void *data)
+   {
+     if (error != SOFTAP_ERROR_NONE) {
+       if (code != SOFTAP_DISABLED_BY_REQUEST)
+         return;
+
+       printf("Failed to disable SoftAP[0x%X]", error);
+       return;
+   }
+     printf("SoftAP is disabled.");
+   }	
+   ```
+
+To register the callback:	
+   ```
+   ret = softap_set_enabled_cb(softap, enabled_cb, NULL);                                 
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Fail to regester the callback.");
+   ``` 
+
+## Handling SoftAP client
+
+To handle SoftAP client:
+
+1. Define and register `client_connection_state_changed_cb()`. It is called when client is connected or disconnected. 
+   ```
+   static void cliet_connection_state_changed_cb(softap_client_h client, bool open, void *data)
+   {
+     softap_client_h clone = NULL;
+     char *ip_address = NULL;
+     char *mac_address = NULL;
+     char *hostname = NULL;
+
+     softap_client_clone(&clone, client);
+     if (clone == NULL) {
+       printf("Failed to clone client handle");
+       return;
+     }
+
+     softap_client_get_ip_address(clone, SOFTAP_ADDRESS_FAMILY_IPV4, &ip_address);
+     softap_client_get_mac_address(clone, &mac_address);
+     softap_client_get_name(clone, &hostname);
+
+     if (open) {
+       printf("New station IP [%s], MAC [%s], hostname [%s]",
+                                   ip_address, mac_address, hostname);
+     } else {
+       print("Disconnected station IP [%s], MAC [%s], hostname [%s]\n",
+                                           ip_address, mac_address, hostname);
+     }
+
+     if (ip_address)
+       free(ip_address);
+     if (mac_address)
+       free(mac_address);
+     if (hostname)
+       free(hostname);
+
+     softap_client_destroy(clone);
+   }
+   ```
+
+To register the callback:
+   ```
+   ret = softap_set_client_connection_state_changed_cb(softap, client_connection_state_changed_cb, NULL);
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Fail to regater the callback.");
+   ```
+
+2. Get the list of all connected SoftAP clients.
```

Comment:
Please enter a line, or add 2 spaces after this sentence to make a line break.
In md file, just a line break is considered as a paragraph.

![image](https://user-images.githubusercontent.com/9958374/44975084-7a838a00-af9c-11e8-8889-8b38a7e31426.png)